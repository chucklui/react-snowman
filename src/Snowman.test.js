import { render, fireEvent } from "@testing-library/react";
import Snowman from "./Snowman";

/** Smoke Test */
it("works at all", function() {
  render(
    <Snowman />
  );
});

/** Test losing state */

it("shows end game state on loss", function() {
    const { container } = render(
      <Snowman maxGuesses={4} />
    );

    const buttonB = container.querySelector("button[value='b']");
    fireEvent.click(buttonB);
    const buttonZ = container.querySelector("button[value='z']");
    fireEvent.click(buttonZ);
    const buttonM = container.querySelector("button[value='m']");
    fireEvent.click(buttonM);
    const buttonY = container.querySelector("button[value='y']");
    fireEvent.click(buttonY);
    // const buttonX = container.querySelector("button[value='x']");
    // fireEvent.click(buttonX);
    // const buttonO = container.querySelector("button[value='o']");
    // fireEvent.click(buttonO);

    // expect the first image to show, but not the second
    expect(container
            .querySelector('.Snowman>img')
            .getAttribute('alt')
        ).toEqual('4');
    expect(
      container.querySelector('.Snowman-buttons')
    ).not.toBeInTheDocument();
    expect(
      container.querySelector('.Snowman-endmessage')
    ).toBeInTheDocument();
    
    //snapshot
    expect(container).toMatchSnapshot();
  });