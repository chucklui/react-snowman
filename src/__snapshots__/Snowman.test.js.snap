// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shows end game state on loss 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="4"
      src="4.png"
    />
    <p
      class="Snowman-wrongGuesses"
    >
      Number wrong: 
      4
    </p>
    <p
      class="Snowman-word"
    >
      _
      _
      _
      _
      _
    </p>
    <p
      class="Snowman-endmessage"
    >
      You Lose, the correct word is 
      apple
    </p>
  </div>
</div>
`;
